lib/docusign_dtr/query_param_helper.rb

```diff
@@ -0,0 +1,127 @@
+module DocusignDtr
+  class QueryParamHelper
+    # rubocop:disable Naming/PredicateName
+    QUERY_PARAMS = {
+      search: :search,
+      count: :count,
+      end_date: :endDate,
+      start_position: :startPosition,
+      room_status: :roomStatus,
+      owned_only: :ownedOnly,
+      transaction_side: :transactionSide,
+      is_under_contract: :isUnderContract,
+      region_id: :regionId,
+      office_id: :officeId,
+      has_submitted_task_list: :hasSubmittedTaskList,
+      has_contract_amount: :hasContractAmount,
+      sort: :sort,
+      date_range_type: :dateRangeType,
+      start_date: :startDate
+    }.freeze
+
+    class << self
+      def call(options)
+        @options = options
+        QUERY_PARAMS.each_with_object({}) do |(key, value), memo|
+          memo[value] = send(key) if @options.key? key
+        end
+      end
+
+      def search
+        @options[:search].to_s
+      end
+
+      def count
+        @options[:count].to_i
+      end
+
+      def start_position
+        @options[:start_position].to_i
+      end
+
+      def room_status
+        raise "error: not acceptable value #{@options[:room_status]}" unless acceptable_value?(
+          :room_status, @options[:room_status]
+        )
+
+        @options[:room_status].to_s
+      end
+
+      def owned_only
+        to_boolean(@options[:owned_only])
+      end
+
+      def transaction_side
+        raise "error: not acceptable value #{@options[:transaction_side]}" unless acceptable_value?(
+          :transaction_side, @options[:transaction_side]
+        )
+
+        @options[:transaction_side].to_s
+      end
+
+      def is_under_contract
+        to_boolean(@options[:is_under_contract])
+      end
+
+      def region_id
+        @options[:region_id].to_i
+      end
+
+      def office_id
+        @options[:office_id].to_i
+      end
+
+      def has_submitted_task_list
+        to_boolean(@options[:has_submitted_task_list])
+      end
+
+      def has_contract_amount
+        to_boolean(@options[:has_contract_amount])
+      end
+
+      def sort
+        raise "error: not acceptable value #{@options[:sort]}" unless acceptable_value?(:sort, @options[:sort].to_s)
```

Comment:
can we define a custom error here?
class InvalidParameter < StandardError; end
raise InvalidParameter.new("value #{value} is not valid for #{key}") or something along those lines?

This could be in a separate PR if you wish!

Reply:
I will refactor and cover with tests in a new PR!